packages/testing/src/Diff/ExportedArrayDiffer.php

```diff
@@ -107,76 +108,45 @@ protected function compareArrays(array $before, array $after, bool $could_be_lis
    * @return array|false
    */
   protected function compareLists(array $before, array $after): array|false {
-    $diff = $this->doCompareLists($before, $after);
-    if (!$diff || count($diff) === count($before) + count($after)) {
-      // The two lists are completely different.
+    $algorithm = new MyersDiff();
+    /** @var list<array{mixed, -1|0|1}> $myers_diff_solution */
+    $myers_diff_solution = $algorithm->calculate(
+      // The Myers diff actually works for non-strings.
+      // See https://github.com/fisharebest/algorithm/pull/9.
+      // @phpstan-ignore argument.type
+      $before,
+      // @phpstan-ignore argument.type
```

Comment:
This is no longer needed once this is released:
- https://github.com/fisharebest/algorithm/pull/9